src/OrchardCore/OrchardCore/Extensions/ExtensionManager.cs

```diff
@@ -152,7 +147,111 @@ public IEnumerable<IFeatureInfo> GetDependentFeatures(string featureId)
             })).Value;
         }
 
-        private IEnumerable<IFeatureInfo> GetFeatureDependencies(
+        private void EnsureInitialized()
+        {
+            if (_isInitialized)
+            {
+                return;
+            }
+
+            lock (_synLock)
+            {
+                if (_isInitialized)
+                {
+                    return;
+                }
+
+                var applicationContext = _serviceProvider.GetRequiredService<IApplicationContext>();
+                var typeFeatureProvider = _serviceProvider.GetRequiredService<ITypeFeatureProvider>();
+                var featuresProvider = _serviceProvider.GetRequiredService<IFeaturesProvider>();
+
+                var extensionDependencyStrategies = _serviceProvider.GetServices<IExtensionDependencyStrategy>();
+                var extensionPriorityStrategies = _serviceProvider.GetServices<IExtensionPriorityStrategy>();
+
+                var modules = applicationContext.Application.Modules;
+                var loadedExtensions = new ConcurrentDictionary<string, ExtensionEntry>();
+
+                // Load all extensions in parallel
+                Parallel.ForEach(modules, (module, cancellationToken) =>
+                {
+                    if (!module.ModuleInfo.Exists)
+                    {
+                        return;
+                    }
+
+                    var manifestInfo = new ManifestInfo(module.ModuleInfo);
+                    var extensionInfo = new ExtensionInfo(module.SubPath, manifestInfo, (mi, ei) =>
```

Comment:
what does `mi` and `ei` stand for? maybe better to give them a readable name

Reply:
Done.